error.go

```diff
@@ -0,0 +1,43 @@
+package consul
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+)
+
+// IsNotFound reports whether the error is a not found error.
+func IsNotFound(err error) bool {
+	if nf, ok := err.(notFound); ok {
+		return nf.NotFound()
+	}
+	return false
+}
+
+type notFound interface {
+	NotFound() bool
+}
+
+type baseError struct {
```

Comment:
Is `baseError` a good name this this? How about `httpError` since this is returned when there's an HTTP status code that represents an error.